cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/internal/query/QueryImpl.java

```diff
@@ -137,15 +137,17 @@ public void close() {
      *  @return Map of indexes in the database.
      */
     @Override
-    public List<Index> listIndexes() {
+    public List<Index> listIndexes() throws QueryException {
         try {
             return dbQueue.submit(new ListIndexesCallable()).get();
         } catch (InterruptedException e) {
-            logger.log(Level.SEVERE,"Failed to list indexes",e);
-            throw new RuntimeException(e);
+            String msg = "Failed to list indexes";
```

Comment:
You could consider cribbing `DatastoreImpl.get` which I wrote to help us avoid this repeating pattern:

```java

    // helper to avoid having to catch ExecutionExceptions
    public static <T> T get(Future<T> future) throws ExecutionException {
        try {
            return future.get();
        } catch (InterruptedException e) {
            logger.log(Level.SEVERE, "Re-throwing InterruptedException as ExecutionException");
            throw new ExecutionException(e);
        }
    }

```